api/kyverno/v2alpha1/exception_type.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2022 The Kubernetes authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v2alpha1
+
+import (
+	kyvernov1 "github.com/kyverno/kyverno/api/kyverno/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PolicyException declares resources to be excluded from specified policies.
+type PolicyException struct {
+	metav1.TypeMeta   `json:",inline,omitempty" yaml:",inline,omitempty"`
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	// Spec declares policy exception behaviors.
+	Spec PolicyExceptionSpec `json:"spec" yaml:"spec"`
+}
+
+type PolicyExceptionSpec struct {
+	Exceptions []Exception `json:"exceptions" yaml:"exceptions"`
+
+	Exclude kyvernov1.MatchResources `json:"exclude" yaml:"exclude"`
+}
+
+type Exception struct {
+	// PolicyName defines the excepted policy.
+	PolicyName string `json:"policyName" yaml:"policyName"`
+	// RuleName is a list which contains the target excepted rules.
+	RuleNames []string `json:"ruleNames" yaml:"ruleNames"`
```

Comment:
What happens if 0 rules are declared?

Reply:
Shall the policy exception be rejected ? Apply to nothing ? Or apply to all rules ?

Reply:
In our previous design, the exception would apply to nothing. 

Well, as we'd do the refactoring, and from the user's perspective, maybe apply to all rules make more sense?

Reply:
IMHO apply to nothing is better, more explicit.
We could support wildcard patterns eventually.

Reply:
Yes, I agree. It should apply to nothing if no rules are listed. 

We can also add a validation check for that.